components/mdx/Sandbox.tsx

```diff
@@ -11,39 +12,77 @@ export default function Sandbox({ id }) {
   }>()
 
   useEffect(() => {
-    fetch(`/api/get-sandbox?id=${id}`)
-      .then((rsp) => rsp.json())
-      .then(setData)
-  }, [id])
+    // Wait to load section until in view
+    const sectionObserver = new IntersectionObserver(([entry], observer) => {
+      if (entry.isIntersecting) {
+        // Once observed, don't observe again
+        observer.unobserve(entry.target)
+
+        // Fetch once observed
+        fetch(`/api/get-sandbox?id=${id}`)
+          .then((rsp) => rsp.json())
+          .then(setData)
+      }
+    })
 
-  if (!data) return null
+    sectionObserver.observe(sandboxRef.current)
+
+    return () => {
+      sectionObserver.disconnect()
+    }
+  }, [id])
 
   return (
-    <div>
-      <a href={`https://codesandbox.io/s/${data.alias}`} target="_blank" rel="noreferrer">
-        <Image
-          className="rounded shadow-lg"
-          src={`https://codesandbox.io/api/v1/sandboxes/${id}/screenshot.png`}
-          alt={data.title}
-          width={1763}
-          height={926}
-          loading="lazy"
-        />
-      </a>
-      <h6 className="text-gray-700 font-bold mt-4">{data.title}</h6>
-      <p className="text-gray-700">{data.description}</p>
-      <div className="w-full">
-        {data.tags.map((tag, i) => (
-          <span
-            key={i}
-            className={`inline-block mt-1 text-gray-500 bg-gray-100 rounded px-1 py-1 text-xs ${
-              i !== data.tags.length - 1 ? 'mr-1' : null
-            }`}
-          >
-            {tag}
-          </span>
-        ))}
-      </div>
+    <div ref={sandboxRef}>
+      {/* Render skeleton while loading */}
+      {!data && (
+        <>
+          <div aria-hidden className="loading rounded shadow-lg" style={{ height: 128 }} />
+          <h6 aria-hidden className="loading rounded text-gray-700 font-bold mt-4">
+            loading
+          </h6>
+          <p aria-hidden className="loading rounded text-gray-700 mt-1">
+            loading
+          </p>
+          <div className="w-full">
+            <span
+              aria-hidden
+              className="loading inline-block mt-2 text-gray-500 bg-gray-100 rounded px-1 py-1 text-xs mr-1"
+            >
+              loading
+            </span>
+          </div>
+        </>
+      )}
+      {data && (
```

Comment:
Maybe there's a better way of toggling that also waits for the image to load. With Suspense?